lib/trace/index.js

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var errors = require('../errors');
+var safeParse = require('../util').safeParse;
+var toCamelCase = require('../util').toCamelCase;
+
+var core = require('./core');
+var TChannelTracer = require('./tchannel');
+var LogTracer = require('./log');
+
+function registerTraceEndpoints(ringpop) {
+    // XXX testing only?
+    //
+    // @wolski, we chatted briefly about this -- whatcha think should
+    // be done? the testing framework doesn't work, as is. (I can add
+    // more mocks to make it go.)
+    if (!ringpop || !ringpop.channel || !ringpop.membership) {
+        return;
+    }
+
+    registerTraceCommand(ringpop, '/trace/add', createTracer);
+    registerTraceCommand(ringpop, '/trace/remove', destroyTracer);
+}
+
+function registerTraceCommand(ringpop, endpoint, processor) {
+    var logPrefix = 'ringpop' + endpoint.replace(/\//g, '.');
+
+    ringpop.channel.register(endpoint, function onRequest(req, resp, optsStr) {
+        resp.headers.as = 'raw';
+        var opts = toCamelCase(safeParse(optsStr));
+        var config = core.resolveTracerConfig(ringpop, opts.event);
+        if (!config) {
+            var err = errors.InvalidOptionError({ option: 'event', reason: 'it is invalid'});
+            ringpop.logger.warn(logPrefix + ': invalid event', err);
+            resp.sendNotOk(err);
+            return;
+        }
+
+        processor(ringpop, config, opts, function done(err) {
+            if (err) {
+                ringpop.logger.warn(logPrefix + ': remote error', err);
+                resp.sendNotOk(err);
+                return;
+            } 
+            ringpop.logger.info(logPrefix + ': success', opts);
+            resp.sendOk();
+        });
+    });
+}
+
+var knownTracerTypes = {
```

Comment:
thoughts on pulling this out into `lib/trace/types.js`? It might seem like overkill, but I do think it promotes supported tracer types to a higher-level.